src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/CodeStyle/CSharpCodeStyleOptions_Parsing.cs

```diff
@@ -12,132 +12,85 @@ namespace Microsoft.CodeAnalysis.CSharp.CodeStyle
     internal static partial class CSharpCodeStyleOptions
     {
         public static CodeStyleOption2<ExpressionBodyPreference> ParseExpressionBodyPreference(
-            string optionString, CodeStyleOption2<ExpressionBodyPreference> @default)
+            string optionString, CodeStyleOption2<ExpressionBodyPreference> @default, Func<string, Optional<ExpressionBodyPreference>> editorConfigDataParse)
```

Comment:
could all of these just take in the appropriate value holder instead of this func?  Then call GetValueFromString on it?